node/cn/backend.go

```diff
@@ -265,6 +271,21 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 	}
 	bc.SetCanonicalBlock(config.StartBlockNumber)
 
+	// Write the live pruning flag to database if the node is started for the first time
+	if config.LivePruning && !chainDB.ReadPruningEnabled() {
+		if bc.CurrentBlock().NumberU64() > 0 {
+			return nil, errors.New("cannot enable live pruning after chain has advanced")
```

Comment:
Turning the live pruning on and off does seem possible. But this hypothetical feature has some consistency risks. Suppose a database is built without pruning, where all trie nodes are legacy nodes (i.e. no extension). If we enable the pruning, then obsolete legacy nodes will be deleted (some nodes under block 10000 in your example). Note that we added extensions to safely delete trie nodes, and in contrast, legacy nodes are not safely removable because they may be referenced from multiple tries. So I don't think it is easy to implement such a feature.